hive/lib/src/hive.dart

```diff
@@ -69,6 +69,14 @@ abstract class HiveInterface implements TypeRegistry {
 
   /// Checks if a box exists
   Future<bool> boxExists(String name, {String? path});
+
+  /// Clears all registered adapters.
+  ///
+  /// To register an adapter use [registerAdapter].
+  ///
+  /// NOTE: [resetAdapters] also clears the default adapters registered
+  /// by Hive.
```

Comment:
Can we somehow add a warning that this is only intended to be used in test environments? So we can say "we already warned you" when we got issues caused by accidentally or intentionally resetting adapters causing other side effects.

Reply:
Like:

```suggestion
  /// by Hive.
  ///
  /// WARNING: This method is only intended to be used for integration and unit tests and
  /// SHOULD not be used in production code.
```